src/server/pfs/server/internal_api_server.go

```diff
@@ -104,7 +103,8 @@ func (a *internalAPIServer) DeleteRepo(ctx context.Context, request *pfs.DeleteR
 	if err != nil {
 		return nil, err
 	}
-	if err := a.driver.DeleteRepo(request.Repo, shards); err != nil && err != pfsserver.ErrRepoNotFound {
+	_, isErrRepoNotFound := err.(*pfsserver.ErrRepoNotFound)
+	if err := a.driver.DeleteRepo(request.Repo, shards); err != nil && !isErrRepoNotFound {
```

Comment:
Do these lines make sense. Isn't the error you're checking on line 106 guaranteed to be `nil` when you reach there?